mne/decoding/transformer.py

```diff
@@ -168,35 +169,41 @@ def __init__(self, info=None):
         self.n_channels = None
         self.n_times = None
 
-    def fit(self, epochs_data, y):
+    def fit(self, epochs, y=None):
         """For each epoch, concatenate data from different channels into a single
         feature vector.
 
         Parameters
         ----------
-        epochs_data : array, shape (n_epochs, n_channels, n_times)
+        epochs : instance of Epochs
             The data to concatenate channels.
-        y : array, shape (n_epochs,)
-            The label for each epoch.
+        y : None | array of shape(n_epochs,)
+            The label of each epoch
 
         Returns
         -------
         self : instance of EpochsVectorizer
             returns the modified instance
         """
-        if not isinstance(epochs_data, np.ndarray):
-            raise ValueError("epochs_data should be of type ndarray (got %s)."
-                             % type(epochs_data))
+        from mne import Epochs
+
+        if not isinstance(epochs, Epochs):
+            raise ValueError("epochs should be an instance of Epochs (got %s)"
+                             % type(epochs))
+
+        self.epochs_data = epochs.get_data()
+        # save epochs for inverse transform
+        self._epochs = epochs
```

Comment:
Is this necessary?? `epochs` can be very heavy, we just need its `info`, `tmin` and `events`